File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/ipersistence/IGroupPersistence.java
package it.dani.selfhomeapp.middle.ipersistence;

import androidx.annotation.NonNull;

import java.util.Set;

public interface IGroupPersistence {
    public Set<String> getGroups();
    public boolean setStateGroup(@NonNull String groupName, int state);
    public boolean addNewGroup(@NonNull String groupName);
    public boolean delGroup(@NonNull String groupName);
    public Set<String> getDeviceFromGroup(@NonNull String groupName);
    public boolean addDeviceToGroup(@NonNull String deviceName, @NonNull String groupName);
    public boolean delDeviceToGroup(@NonNull String deviceName, @NonNull String groupName);
}

File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/ipersistence/IGroupPersistence.kt
package it.dani.selfhomeapp.middle.ipersistence

interface IGroupPersistence {
    fun getGroups() : Set<String>
    fun setStateGroup(groupName : String, state : Int) : Boolean
    fun addNewGroup(groupName : String) : Boolean
    fun delGroup(groupName : String) : Boolean
    fun getDeviceFromGroup(groupName : String) : Set<String>
    fun addDeviceToGroup(deviceName : String,groupName : String) : Boolean
    fun delDeviceToGroup(deviceName : String, groupName: String) : Boolean
}